neqo-bin/src/server/mod.rs

```diff
@@ -204,29 +209,34 @@ fn qns_read_response(filename: &str) -> Result<Vec<u8>, io::Error> {
 
 #[expect(clippy::module_name_repetitions, reason = "This is OK.")]
 pub trait HttpServer: Display {
-    fn process_multiple(
+    fn new(
+        args: &Args,
+        anti_replay: AntiReplay,
+        cid_mgr: Rc<RefCell<dyn ConnectionIdGenerator>>,
+    ) -> Self;
+    fn process_multiple<'a>(
         &mut self,
-        dgram: Option<Datagram<&mut [u8]>>,
+        dgrams: impl IntoIterator<Item = Datagram<&'a mut [u8]>>,
```

Comment:
This change makes `HttpServer` no longer [_dyn compatible_](https://doc.rust-lang.org/reference/items/traits.html#dyn-compatibility).

Instead of `Runner` owning a `server: Box<dyn HttpServer>`, it now owns a `server: S` where `S` is a new trait parameter.

In my eyes the additional complexity (trait parameter `S`) is worth it. Dynamic dispatching is difficult to work with in Rust, especially when using Rust's Futures. In addition, given that `neqo-bin` is at the outskirt of our code base, the new trait parameter `S` does not infect other places.